### PR TITLE
refactor(react-base-plugin): jsx generation extracted to shared package

### PR DESCRIPTION
**Pull request by alexnm**

### CODE REVIEW

File: packages/teleport-component-generator-preact/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# teleport-component-generator-react
```
Comment (JayaKrishnaNamburu) [inline packages/teleport-component-generator-preact/README.md]: Name change from react to preact

**Comment by JayaKrishnaNamburu** [comment]
Can we check the support for rest of the styled flavors at once, are we will come back again.

Like for example ``StyledComponents`` have a pretty straight forward implementation like react, we can add that style plugin in ``teleport-component-generator-preact``

**Comment by alexnm** [comment]
> Can we check the support for rest of the styled flavors at once, are we will come back again.
> 
> Like for example `StyledComponents` have a pretty straight forward implementation like react, we can add that style plugin in `teleport-component-generator-preact`

I will create a preact follow-up issue with a couple of things that can be worked on, including more styling options and a project-generator with a basic setup. However, I would prioritize other issues for now, didn't want to stay too much on this

**Comment by JayaKrishnaNamburu** [comment]
> > Can we check the support for rest of the styled flavors at once, are we will come back again.
> > Like for example `StyledComponents` have a pretty straight forward implementation like react, we can add that style plugin in `teleport-component-generator-preact`
> 
> I will create a preact follow-up issue with a couple of things that can be worked on, including more styling options and a project-generator with a basic setup. However, I would prioritize other issues for now, didn't want to stay too much on this

Cool, sounds like a plan 👍

**Pull request merged**